fpx/src/data/migrations/20240708_create_spans.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE spans (
+    trace_id TEXT,
```

Comment:
I noticed this in the requests migration as well, but is there a specific reason we don't enforce things like `NOT NULL` etc here?

Now the model does not actually match the underlying table:
```rust
pub trace_id: String,
pub span_id: String,
pub parent_span_id: Option<String>,
```

You cannot alter table columns in migrations, so you have to create a new table / copy / drop to migrate things like that. Is that the reason?

Reply:
I assumed (maybe wrongfully so) that `NOT NULL` is implicit, whereas `NULL` is not.

I think we should be able to change the columns later on. 

Though both things might work differently on sqlite/libsql.

Reply:
SQLite seems to have no `ALTER COLUMN` or anything similar :( So it looks like the only option is to create a new table and copy stuff over.

And columns seem to be nullable be default as well

Reply:
> SQLite seems to have no `ALTER COLUMN` or anything similar :( So it looks like the only option is to create a new table and copy stuff over.
> 
> And columns seem to be nullable be default as well

Thanks for figuring these things out. I'll change everything to have a `NOT NULL`. Regarding the column modification, that sounds like migrations are going to be extra annoying if we want to add a column...